boot/bootutil/src/bootutil_misc.c

```diff
@@ -107,6 +108,38 @@ static const struct boot_swap_table boot_swap_tables[] = {
 #define BOOT_SWAP_TABLES_COUNT \
     (sizeof boot_swap_tables / sizeof boot_swap_tables[0])
 
+/**
+ * @brief Determine if the data at two memory addresses is equal
+ *
+ * @param s1    The first  memory region to compare.
+ * @param s2    The second memory region to compare.
+ * @param n     The amount of bytes to compare.
+ *
+ * @note        This function does not comply with the specification of memcmp,
+ *              so should not be considered a drop-in replacement.
+ *
+ * @return      FIH_SUCCESS if memory regions are equal, otherwise FIH_FAILURE
+ */
+fih_int boot_secure_memequal(const void *s1, const void *s2, size_t n)
```

Comment:
I can see what you're doing but when I see `secure_memequal` I expect a function that is constant time, which is not the case here, maybe a more appropriate name would be helpful, maybe settle on `fih` like `boot_fih_memequal` (sorry for the lack of inspiration!)